### PR TITLE
Improve Integrated Auth Failure message with SQL Server

### PR DESCRIPTION
**Pull request by saurabh500**
When an integrated auth failure occurs, then an error message pointing to running kinit is shown. Based on the experiences of Sql Ops Studio users, it seems like lot of customers don't have an SPN registered for the SQL Server to which they are trying to connect to, hence a Kerberos token for Sql Server cannot be generated correctly from the KDC and the authentication cannot be performed correctly. The error message should hint the users to check for both kinit command on the client side and to check the server configuration to make sure that an SPN is registered. 

Fixes https://github.com/dotnet/corefx/issues/25328

### CODE REVIEW

**Comment by geleems** [comment]
LGTM

**Comment by saurabh500** [comment]
@kburtram I am adding to the Integrated Auth error message in this PR. From the issues that I am seeing, the customers don't have an SPN registed on Sql Server which causes failures. If you think anything else needs to be added to the error message, then let me know.

**Comment by saurabh500** [comment]
@dotnet-bot Test Alpine.3.6 x64 Debug Build please
Test Windows x86 Release Build please
Test Windows x64 Debug Build please

**Comment by saurabh500** [comment]
@dotnet-bot Test Alpine.3.6 x64 Debug Build please
test Linux x64 Release Build please
Test OSX x64 Debug Build please

File: src/System.Data.SqlClient/src/Resources/Strings.resx [hunk 1]
```diff
@@ -431,8 +431,7 @@
     <value>Failed to generate SSPI context.</value>
   </data>
   <data name="SQL_KerberosTicketMissingError" xml:space="preserve">
-    <value>Cannot access Kerberos ticket. Ensure Kerberos has been initialized with 'kinit'.</value>
-  </data>
+    <value>Cannot authenticate using Kerberos. Ensure Kerberos has been initialized on the client with 'kinit' and a Service Principal Name has been registered for the SQL Server to allow Kerberos authentication.</value>  </data>
```
Comment (stephentoub) [inline src/System.Data.SqlClient/src/Resources/Strings.resx]: Nit: formatting, `</data>` got pulled up to the previous line.

**Pull request merged**